tests/rptest/services/producer_swarm.py

```diff
@@ -121,3 +130,44 @@ def wait_node(self, node, timeout_sec=600):
 
     def stop_node(self, node):
         node.account.ssh(f"bash /opt/remote/control/stop.sh {self.EXE}")
+
+    def _remote_url(self, node, path) -> str:
+        return f"http://{node.account.hostname}:{self._remote_port}/{path}"
+
+    def _get(self, node, rest_handle):
+        """
+            Perform a GET to client_swarm's metrics API.
+        """
+        url = self._remote_url(node, rest_handle)
+        try:
+            r = requests.get(url, timeout=10)
+        except Exception as e:
+            raise RuntimeError(f"Failed to get data from '{url}'") from e
+        return r.json()
+
+    @dataclass
+    class MetricsSummary:
+        p0: int
+        p50: int
+        p100: int
+
+    def get_metrics_summary(self, seconds=None) -> MetricsSummary:
+        path = f"metrics/summary"
+        if seconds:
+            path = f"{path}?seconds={seconds}"
+
+        res = self._get(self._node, path)
+
+        return self.MetricsSummary(int(res["min"]), int(res["median"]),
+                                   int(res["max"]))
+
+    def await_progress(self, target_msg_rate, timeout_sec, err_msg=None):
+        time.sleep(0.5)
```

Comment:
this seems odd?

Reply:
Nice catch, that was an artifact from testing. I've removed it in the latest rev.